### PR TITLE
configuration-reload-test: wait for the new worker to get ready

### PR DESCRIPTION
**Pull request by oschaaf**
After sending a SIGHUP, wait for the new worker process with the
new configuration to handle further requests, or else the tests
running after that might get confused when the new worker becomes
active with fresh shared mem statistics and shared mem cache.

### CODE REVIEW

File: test/nginx_system_test.sh [hunk 1]
```diff
@@ -551,6 +553,11 @@ fire_ab_load
 check wget $EXAMPLE_ROOT/styles/W.rewrite_css_images.css.pagespeed.cf.Hash.css \
   -O /dev/null
 check_simple "$NGINX_EXECUTABLE" -s reload -c "$PAGESPEED_CONF"
+# Wait for the new worker process with the new configuration to get ready, or
+# else the sudden reset of the shared mem statistics/cache might catch upcoming
+# tests unaware.
+fetch_until $STATISTICS_URL \
+  'grep -c image_rewrite_total_original_bytes:[[:space:]]*0' 1
```
Comment (jeffkaufman) [inline test/nginx_system_test.sh]: Don't we have a scrape_stat function for this?

**Comment by jeffkaufman** [comment]
This looks great!  I think this might fix the intermittent cache_flush test error we've seen.

(Specifically, we'd see the cache flush properly, but the stat wouldn't get bumped.  That could be explained by the fresh stats coming online at exactly the wrong time, so that the flush happened on the old stats and the query to the stats url happened on the new stats.  This was especially likely because that test has some sleeps.)

File: test/nginx_system_test.sh (hunk 1 above)
Comment (oschaaf) [inline test/nginx_system_test.sh]: Re: "Don't we have a scrape_stat function for this?"
Yes - updated. PTAL?

**Comment by oschaaf** [comment]
@jeffkaufman "I think this might fix the intermittent cache_flush test error we've seen."
That sounds like something this could very well fix. I've only seen that flake one or two times in the past weeks, so it's hard to be sure.

**Comment by jeffkaufman** [comment]
LGTM!

**Comment by oschaaf** [comment]
@jeffkaufman Ugh. With this merged, the sanity check for `error.log` starts failing when running the tests in a loop under valgrind. Inspecting `error.log`, it looks like somehow the old worker process isn't able to finish up all queued requests in the timeout period. However, it does not look like that happens because it's short on time. Some requests from `ab` just seem to stall, so I think I need to look at if there's an edge case that I missed.

**Pull request merged**